src/cache/prognostic_edmf_precomputed_quantities.jl

```diff
@@ -411,6 +268,27 @@ NVTX.@annotate function set_prognostic_edmf_precomputed_quantities_explicit_clos
             dt,
         )
 
+        # If the surface buoyancy flux is positive, increase entrainment in the first cell
+        # so that the updraft area grows to at least `surface_area` within one timestep.
+        # Otherwise (stable surface), leave entrainment unchanged.
+        buoyancy_flux_val = Fields.field_values(p.precomputed.sfc_conditions.buoyancy_flux)
+        sgsʲs_ρ_int_val = Fields.field_values(Fields.level(ᶜρʲs.:($j), 1))
+        sgsʲs_ρa_int_val =
+            Fields.field_values(Fields.level(Y.c.sgsʲs.:($j).ρa, 1))
+        @. ᶜaʲ_int_val = draft_area(sgsʲs_ρa_int_val, sgsʲs_ρ_int_val)
+        entr_int_val = Fields.field_values(Fields.level(ᶜentrʲs.:($j), 1))
+        detr_int_val = Fields.field_values(Fields.level(ᶜdetrʲs.:($j), 1))
+        @. entr_int_val = limit_entrainment(
+            ifelse(
+                buoyancy_flux_val < 0 || ᶜaʲ_int_val >= $(FT(turbconv_params.surface_area)),
+                entr_int_val,
+                detr_int_val +
+                ($(FT(turbconv_params.surface_area)) / ᶜaʲ_int_val - FT(1)) / FT(dt),
```

Comment:
Add an eps to ᶜaʲ_int_val to be safe? Also do you need the FTs in this function?